src_c/line.c

```diff
@@ -490,6 +490,89 @@ pg_line_flip_ip(pgLineObject *self, PyObject *_null)
     Py_RETURN_NONE;
 }
 
+double 
+_lerp_helper(float start, float end, float amount) {
+    return (1 - amount) * start + amount * end;
+}
+
+static int
+_line_scale_helper(pgLineObject *line, double factor, double origin) {
+    if (factor == 1.0) {
+        return 1;
+    }
+    else if (factor == 0.0) {
+        PyErr_SetString(PyExc_ValueError, "Cannot scale by 0");
+        return 0;
+    }
+    else if (factor < 0.0) {
+        PyErr_SetString(PyExc_ValueError, "Cannot scale by a negative number");
+        return 0;
+    }
+
+    if (origin < 0.0) {
+        PyErr_SetString(PyExc_ValueError, "Origin cannot be negative");
+        return 0;
+    }
+
+    double x1 = line->line.x1;
+    double y1 = line->line.y1;
+    double x2 = line->line.x2;
+    double y2 = line->line.y2;
+
+    double x1_factor = x1 * factor;
+    double y1_factor = y1 * factor;
+    double x2_factor = x2 * factor;
+    double y2_factor = y2 * factor;
+
+    double dx = _lerp_helper(x1_factor - x1, x2_factor - x2, origin);
+    double dy = _lerp_helper(y1_factor - y1, y2_factor - y2, origin);
+    
+    line->line.x1 = x1_factor - dx;
+    line->line.y1 = y1_factor - dy;
+    line->line.x2 = x2_factor - dx;
+    line->line.y2 = y2_factor - dy;
+
+    return 1;
+}
+
+static PyObject *
+pg_line_scale(pgLineObject *self, PyObject *const *args, Py_ssize_t nargs) {
+    double factor;
+    double origin;
+
+    if (!pg_TwoDoublesFromFastcallArgs(args, nargs, &factor, &origin)) {
+        return NULL;
+    }
+
+    PyObject *line;
+    if (!(line = pgLine_New(&self->line))) {
+        return NULL;
+    }
+
+    if (!_line_scale_helper((pgLineObject*)line, factor, origin)) {
```

Comment:
Here you can use the `pgLine_AsLine` macro (if you modify the function as i requested).